Add missing Query API in docs and C header

batiati opened the pull request:
- Update "requests" page with `query_*` operations.
- Update C header with query filter/flags

cb22 commented on docs/README.md:
```diff
@@ -59,6 +59,6 @@ Learn more about TigerBeetle's [mission, history](./about/README.md),
 
 - [𝕏](https://twitter.com/tigerbeetledb)
 - [GitHub](https://github.com/tigerbeetle/tigerbeetle)
-- [Slack](https://slack.tigerbeetle.com/invite)
```
Hmm - I think the idea was that we could keep this constant and update it there without spamming git?

batiati replied:
That's true, I didn't notice it was a `tigerbeetle.com` url. :+1:

batiati committed:
Include missing Query API
- Missing query operations in the docs
- Missing query filter/flags in the C header.
